src/modules/FSXARemoteApi.ts

```diff
@@ -405,8 +404,24 @@ export class FSXARemoteApi implements FSXAApi {
       },
       new Logger(this._logLevel, 'CaaSMapper')
     )
-    const mappedElement = await mapper.mapElementResponse(responseJSON, filterContext)
-    if (!this._caasItemFilter) return mappedElement
+    let { mappedItems, referenceMap, resolvedReferences } = await mapper.mapElementResponse(
+      unmappedElement,
+      additionalParams,
+      filterContext
+    )
+
+    if (denormalized)
+      mappedItems = CaaSMapper.denormalizeResolvedReferences(
+        mappedItems,
+        referenceMap,
+        resolvedReferences
+      )
+
+    // TODO: fix type converting item as Dataset in multiple spots...
+    const mappedElement = mappedItems[0]
+
+    if (!this._caasItemFilter)
+      return denormalized ? mappedElement : { mappedItems, referenceMap, resolvedReferences }
 
     const filteredElement = (
       await this._caasItemFilter({
```

Comment:
This function expects denormalized data by convention. We should talk about how to handle that. 
One could denormalize only for this function, apply it to the normalized result and then return that.